File: Cpp/0409_longest_palindrome.cpp
// https://leetcode.cn/problems/longest-palindrome/

class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> cnt;
        for (char ch: s) {
            cnt[ch] = cnt[ch] + 1;
        }

        int res = 0, odd = 0;
        for (auto it = cnt.begin(); it != cnt.end(); ++it) {
            int count = it->second;
            res = res + count / 2 * 2;
        }
        if (res < s.size()) {
            res = res + 1;
        }
        return res;
    }
};

File: Python3/0409_longest_palindrome.py
# https://leetcode.cn/problems/longest-palindrome/

class Solution:
    def longestPalindrome(self, s):
        """
        Inputs:
            s: str
        Outputs:
            res: int
        """

        cnt = dict()
        for ch in s:
            cnt[ch] = cnt.get(ch, 0) + 1

        res = 0
        for ch in cnt:
            res = res + cnt[ch] // 2 * 2
        if res < len(s):
            res = res + 1
        return res